cmd/dlt/oidcprovider/cmd.go

```diff
@@ -0,0 +1,222 @@
+			"auto: OIDC provider will be deleted using the current AWS account\n"+
+			"manual: Command to delete the OIDC provider will be output",
+	)
+	Cmd.RegisterFlagCompletionFunc("mode", modeCompletion)
+	confirm.AddFlag(flags)
+}
+
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	if !ocm.IsValidClusterKey(clusterKey) {
+		reporter.Errorf(
+			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+		os.Exit(1)
+	}
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+	mode := args.mode
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "OIDC provider deletion mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  mode,
+			Options:  modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid OIDC provider deletion mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading cluster '%s'", clusterKey)
+	clusters, err := ocmClient.GetArchivedCluster(clusterKey)
+	// if err is nil or clusters list is empty or nil we check for cluster in "active" state
+	if err != nil || len(clusters) == 0 || (len(clusters) == 1 && clusters[0] == nil) {
+		c, err := ocmClient.GetCluster(clusterKey, creator)
+		if err != nil {
+			reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+			os.Exit(1)
+		}
+		if c != nil && c.ID() != "" {
+			reporter.Errorf("Cluster '%s' is in '%s' state. Operator roles can be deleted only for the "+
+				"uninstalled clusters", c.ID(), c.State())
+			os.Exit(1)
+		}
+
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
```

Comment:
We should attempt finding the cluster before we ask for the interactive `--mode` - otherwise the user is required to take action before we know whether it will succeed